com.unity.shadergraph/Editor/Serialization/JsonObject.cs

```diff
@@ -14,6 +14,12 @@ public class JsonObject : ISerializationCallbackReceiver
         [SerializeField]
         string m_ObjectId = Guid.NewGuid().ToString("N");
 
+        internal string ObjectIdUnsafe // allow id to be overwritten when necessary
```

Comment:
I would probably make this a function (and remove Unsafe), something like:  OverrideObjectID(...)
So it's more clear the intended use of this path.
No reason for someone to use this property to read the object id.